### PR TITLE
Pi-Hole < 4.3.3 dhcp static address RCE 

### PR DESCRIPTION
**Pull request by h00die**
This adds the last of 3 Pi-Hole exploit modules.  While there are many versions of overlap between this one and the 4.4 one, the benefit here is that a port 80 inbound connection isn't required.  This may help in some cases for exploitation.

This module exploits the DHCP server portion when adding a static reservation.  The DHCP server doesn't need to be running.  There are many constraints on exploitation, if you're interested read the original vuln writeup, it's pretty awesome how the author made it happen.

Also added `rand_text_hex`, it was already in `rex.text`, and I needed it for random MAC addresses.

## Verification

- [ ] Start `msfconsole`
- [ ] `use exploit/unix/http/pihole_dhcp_mac_exec`
- [ ] `set rhosts/lhost/lport`
- [ ] **Verify** you get a userland shell
- [ ] **Document** spelling and grams looks bueno

### CODE REVIEW

**Comment by ccondon-r7** [comment]
Hey @h00die question for you, and maybe a silly one...it seems like the 4.4 RCE module #13445  sort of encompasses the other two Pi-Hole PRs you've got up. Is there a reason to have three instead of just the one? Just trying to figure out where we get the most bang for our buck in terms of value and maintenance surface area!

**Comment by h00die** [comment]
Yeppers!
I tried to capture the difference with this one at least in the first paragraph.

The 4.4 exploit has the advantage of giving a root shell, which is hawt.  However, it REQUIRES the server to connect to MSF over port 80 multiple times (4 in most cases).  To a SOC or other presence, this is rather noisy.  It's also passing back shell code over unencrypted channels and is highly likely to get flagged as suspicious by snort/IDS.  It drops files on disk, which sucks but is unavoidable.  Lastly, it makes a config change which may have unavoidable consequences.  Since I wasn't running this version in production, I'm unsure if there are side effects or not.  They're most likely minimal (some more/different sites may be blocked by the pihole), but its possible.  Cleanup was also not something I was willing to do since it wasn't clean (running `sed` seemed like it could get bad).

The 4.3.3 exploit has a disadvantage of giving a user level shell.  However, it doesn't require a the back and forth port 80 connections of the 4.4 exploit.  It also doesn't put any extra (payload aside) files on disk.  I suspect most people don't use pihole as a DHCP server, so the config change that happens would not have a side effect (although very SMALL chance of a MAC collision). Plus cleanup is trivial and done.

The 3.3 exploit has a disadvantage of giving a user level shell.  It is similar in nature to the 4.3.3 exploit in that it doesn't have the 4.4 port 80 reqs, writing to disk, and all that.  The 3.3 exploit is even less likely (if not impossible) to have config change side effects since we're adding something to the whitelist, not a network wide service change.  Although a cleanup isn't performed, its unlikely the change would have a real world effect anyways.

So mainly we're looking at 4.3.3 & 3.3 are stealthier and don't have a port 80 outbound requirement, plus don't write extra (payload aside) files to disk, and are less likely to cause a noticeable network/system change.

File: modules/exploits/unix/http/pihole_dhcp_mac_exec.rb [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/unix/reverse'
+        },
+        'Payload' => {
+          'BadChars' => "\x00"
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION]
```
Comment (wvu-r7) [inline modules/exploits/unix/http/pihole_dhcp_mac_exec.rb:49]: https://github.com/rapid7/metasploit-framework/pull/13470#issuecomment-630238856

```suggestion
          'Reliability' => [REPEATABLE_SESSION],
          'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
```

**Comment by h00die** [comment]
@wvu-r7 saw your comment, i guess you deleted it.  Wanted to at least note my thought process on why I didn't include `CONFIG_CHANGES`.

I went back and forth with adding `CONFIG_CHANGES`.  It *does* change the config by adding a new mac entry.  However, since the DHCP server most likely isn't running, it doesn't *really* change it.  Plus since we have automated cleanup, it would only change for <60sec.  Decided to leave it off since its changed for such a short time, it gets reverted, and most likely the change isn't applicable since the service isn't running.

**Comment by h00die** [comment]
@todb-r7 after talking with @jbarnett-r7 on slack, is there any way to **help** push this CVE through from it's reserved state? https://github.com/CVEProject/cvelist/blob/master/2020/8xxx/CVE-2020-8816.json

File: documentation/modules/exploit/unix/http/pihole_dhcp_mac_exec.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+### Setup
+
+Install Pi-Hole [Pi-Hole 4.3](https://github.com/pi-hole/pi-hole/releases/tag/v4.3)
+with the following commands:
+
+  ```
+  sudo git clone --depth=1 -b v4.3 https://github.com/pi-hole/pi-hole.git /etc/.pihole
+  # replace 'git clone' with 'git clone -b v4.3'
+  sudo nano /etc/.pihole/automated\ install/basic-install.sh
+  sudo ./basic-install.sh
```
Comment (adfoster-r7) [inline documentation/modules/exploit/unix/http/pihole_dhcp_mac_exec.md:24]: Is this:

```
sudo /etc/.pihole/automated\ install/basic-install.sh
```

Comment (h00die) [reply documentation/modules/exploit/unix/http/pihole_dhcp_mac_exec.md:24]: yes, yes it is!

**Comment by adfoster-r7** [comment]
## Release notes

This exploits a command execution in Pi-Hole <= 4.3.2. A new DHCP static lease is added with a MAC address which includes an RCE. DHCP server is not required to be running.

**Pull request merged**